src/markdown-pages/build-apps/build-visualization.mdx

```diff
@@ -0,0 +1,210 @@
+
+```bash
+nr1 create --type nerdpack --name my-awesome-nerdpack
+```
+
+  </Step>
+
+  <Step>
+
+Navigate to the root of your Nerdpack project, and then run the `create` command.
+
+```bash
+cd my-awesome-nerdpack
+nr1 create --type visualization --name my-awesome-visualization
+```
+
+As a result, you have a new visualization directory matching the name you gave your visualization. It is located in the `visualizations` directory in the root of your nerdpack project.
+
+<img
+  src="../../images/build-an-app/visualization-boilerplate-files.png"
+  alt="Visualization files generated by CLI"
+/>
+
+The files created provide an example visualization – a radar chart populated by a basic NRQL query.
+
+The `nr1.json` file provides metadata including the `configuration` field that defines the prop input fields to be shown in the UI. These are the fields users will fill in to create an instance of the visualization. Supported input types are `boolean`, `string`, `number`, `json`, `nrql`, `namespace`, and `collection`. A `namespace` type groups a set of props together in the prop editing UI under a heading. A `collection` defines a set of fields that can be repeated and result in an array prop type. The `nrqlQueries` entry is an example of a collection type.
+
+The `index.js` file is where you define the React component that receives the props and renders the visualization. You can install and import any javascript libraries you find useful. The example component imports [recharts](https://recharts.org), a library that wraps [D3](https://d3js.org) submodules for easy to compose visualizations.
+
+If your visualization queries data over a time range and you want it to utilize the platform time range setting, you will need to utilize the `timeRange` value from [PlatformStateContext](/components/platform-state-context). The `timeRange` can be passed directly to the [NrqlQuery](/components/nrql-query) component. Setting the NrqlQuery component's `pollInterval` prop to `NrqlQuery.AUTO_POLL_INTERVAL` automatically handles calculating an appriate data polling interval.
+
+```js
+<PlatformStateContext.Consumer>
+  {({timeRange}) => (
+    <NrqlQuery
+      timeRange={timeRange}
+      pollInterval={NrqlQuery.AUTO_POLL_INTERVAL}
+    >
+    ...
+```
+
+  </Step>
+
+</Steps>
+
+## Render the visualization in local development
+
+Now you run the generated visualization locally, and view it in Custom Visualizations, where you can quickly test changes to your code.
+
+<Steps>
+  <Step>
+
+Start the local node server for your Nerdpack.
+
+```bash
+nr1 nerdpack:serve
+```
+
+<Callout variant="important">
+
+To see changes to the nr1.json file take affect, restart the local node server. For example, changes to the definition of the configuration field will not show up in the prop configuration side bar of the Custom Visualizations UI until you stop the local node server and start it back up.
```

Comment:
```suggestion
To see changes to the nr1.json file take affect, restart the local node server. Changes to the definition of the configuration field will not show up in the prop configuration side bar of the Custom Visualizations UI until you stop the local node server and start it back up.
```

Reply:
Good suggestion. It's not letting me commit via your comment because of an earlier commit conflicting but I made the change and pushed it up just now.